app/models/time_slot.rb

```diff
@@ -19,6 +19,9 @@ class TimeSlot < ApplicationRecord
   end
   scope :grid_order, -> { joins(:room).order(:conference_day, :start_time, 'rooms.grid_position') }
 
+  scope :scheduled, -> { where.not(program_session: nil) }
```

Comment:
A scheduled time slot would also include time slots that have custom text in the title.